lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,39 +9,65 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache.
+	 * It can be cleared by calling wp_theme_has_theme_json_clean_cache().
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
+	function wp_theme_has_theme_json() {
+		$cache_group       = 'theme_json';
+		$cache_key         = 'wp_theme_has_theme_json';
```

Comment:
I wanted to note here that this deliberately uses the function name as the cache key for a reason: this is expected to be unique in the PHP namespace.

Additionally, unlike other functions, this is prefixed by `wp_` and not `gutenberg_` as well. If there is ever a need to make this method work differently in the plugin and in core (which requires renaming it to `gutenberg_`), we should also rename the cache key. Otherwise, the plugin will use the cache contents from core if there are there.